jetty-server/src/main/java/org/eclipse/jetty/server/ForwardedRequestCustomizer.java

```diff
@@ -461,81 +461,88 @@ public void customize(Connector connector, HttpConfiguration config, Request req
 
         // Do a single pass through the header fields as it is a more efficient single iteration.
         Forwarded forwarded = new Forwarded(request, config);
+        boolean match = false;
         for (HttpField field : httpFields)
         {
             try
             {
                 MethodHandle handle = _handles.get(field.getName());
                 if (handle != null)
+                {
+                    match = true;
                     handle.invoke(forwarded, field);
+                }
             }
             catch (Throwable t)
             {
                 onError(field, t);
             }
         }
 
-        String proto = "http";
-
-        // Is secure status configured from headers?
-        if (forwarded.isSecure())
+        if (match)
```

Comment:
I don't think this match logic should be required.   There are many headers and some are orthogonal to others, so the logic below must be robust and can't be gated by a match, which may be for an unrelated header.

In the OPs case, proto must not be changed unless there is an explicit header that changes it - either giving a proto or forcing secure.

Perhaps this match is an optimisation to avoid several ifs below, but I don't think it is necessary as this customizer is typically only deployed when all request will have such a header.

Reply:
The match logic here (in jetty-9.4.x) was taken from jetty-10.0.x.

https://github.com/eclipse/jetty.project/blame/jetty-10.0.x/jetty-server/src/main/java/org/eclipse/jetty/server/ForwardedRequestCustomizer.java#L464-L482

The idea behind it is that if there's no Forwarding headers (even those not-quite-forwarding headers `X-Proxied-Https`, `Proxy-auth-cert`, and `Proxy-ssl-id` that ForwardingRequestCustomizer also pays attention to) then there should be no action taken by ForwardingRequestCustomizer, as the incoming request is not based on forwarding.